openstf/model/split_energy.py

```diff
@@ -42,23 +44,65 @@ def split_energy(pid):
     error = components[["load", "Inschatting"]].diff(axis=1).iloc[:, 1]
     mae = error.abs().mean()
     coefdict.update({"MAE": mae})
+    coefsdf = convert_coefdict_to_coefsdf(pj, input_split_function, coefdict)
 
-    # Get average coefs of previous runs
+    # Get average coefs of previous runs and check if new coefs are valid
     mean_coefs = db.get_energy_split_coefs(pj, mean=True)
-    # Loop over keys and check if the difference with the average value is not more than 100%
-    # In case the difference is more tha 100% of the average set KPI coefs as expected to False
+    if are_new_coefs_valid(coefdict, mean_coefs) is False:
+        # If coefs not valid, do not update the coefs in the db and send teams
+        # message that something strange is happening
+        monitoring.post_teams_alert(
+            "New splitting coefficients for pid {} deviate strongly from previously stored coefficients.".format(
+                pj["id"],
+            ),
+            coefsdf=coefsdf,
+        )
+        # Use the last known coefficients for further processing
+        last_coefdict = db.get_energy_split_coefs(pj)
+        last_coefsdf = convert_coefdict_to_coefsdf(
+            pj, input_split_function, last_coefdict
+        )
+        return last_coefsdf
+
+    # Save Results
+    db.write_energy_splitting_coefficients(coefsdf, if_exists="append")
+    logger.info("Succesfully wrote energy split coefficients to database", pid=pj["id"])
+    return coefsdf
+
+
+def are_new_coefs_valid(new_coefs, mean_coefs):
+    """Check if new coefficients are valid.
+
+    Args:
+        new_coefs (dict): new coefficients for standard load profiles
+        mean_coefs (dict): average of previous coefficients for standard load profiles
+
+    Returns:
+        boolean: bool whether new coefs are valid
+    """
+    # Loop over keys and check if the absolute difference with the average value is not
+    # more than COEF_MAX_PCT_DIFF x absolute average value.
     # If no previous coefs are stored an mean_coefs is empty and this loop wil not run
     for key in mean_coefs.keys():
```

Comment:
What should happen if the previous coefs are empty? Can you check explicitly for this and make the desired outcome happen?

Also; what do you want to happen when the keys of old_coefs do not match the new coefs? I suppose you just want to skip that specific key, right? Can you include this to make this function more robust?

Reply:
1. If previous coefs are empty, there is nothing to compare against and I would assume the new coefficients are always good, so they are added to the db. Why add it explicitly?
2. Good one, I added a check for `key in new_coefs`.

Reply:
2.b if key is not present in new coefs, it should fail, because there is an old coef not in the new one